tests/helpers/node_pool.go

```diff
@@ -0,0 +1,193 @@
+func AcquireNodeFromPool(ctx context.Context, t *testing.T, client klient.Client, expiryDuration time.Duration) string {
+	t.Helper()
+
+	nodePoolMutex.Lock()
+	defer nodePoolMutex.Unlock()
+
+	var selectedNodeName string
+
+	require.Eventually(t, func() bool {
+		nodes, err := GetAllNodesNames(ctx, client)
+		if err != nil {
+			t.Logf("failed to get nodes from cluster: %v", err)
+			return false
+		}
+
+		if len(nodes) == 0 {
+			t.Log("no nodes found in cluster")
+			return false
+		}
+
+		for _, nodeName := range nodes {
+			if isNodeAvailable(ctx, t, client, nodeName, expiryDuration) {
+				selectedNodeName = nodeName
+				return true
+			}
+		}
+
+		t.Logf(
+			"No available nodes found, waiting for one to expire or be released. Current time: %s",
+			time.Now().Format(time.RFC3339),
+		)
+
+		return false
+	}, EventuallyWaitTimeout, WaitInterval, "timed out waiting to acquire a node from the pool")
+
+	require.NotEmpty(t, selectedNodeName, "failed to acquire a node")
+
+	err := labelNodeAsUsed(ctx, client, selectedNodeName, t.Name())
+	require.NoError(t, err, "failed to label node %s as used", selectedNodeName)
+
+	t.Logf("Acquired node '%s' for test '%s'", selectedNodeName, t.Name())
+
+	return selectedNodeName
+}
+
+// isNodeAvailable checks if a node is currently available for use by a test.
+// A node is available if it does not have the NodeUsedByLabel, or if the label
+// indicates it was used by an expired test.
+func isNodeAvailable(
+	ctx context.Context,
+	t *testing.T,
+	client klient.Client,
+	nodeName string,
+	expiryDuration time.Duration,
+) bool {
+	node, err := GetNodeByName(ctx, client, nodeName)
+	if err != nil {
+		return false
+	}
+
+	if node.Spec.Unschedulable {
+		return false
+	}
+
+	usedBy, hasUsedBy := node.Labels[NodeUsedByLabel]
+	usedFromStr, hasUsedFrom := node.Labels[NodeUsedFromLabel]
+
+	if !hasUsedBy || !hasUsedFrom {
+		return true
+	}
+
+	var usedFromUnix int64
+
+	_, parseErr := fmt.Sscanf(usedFromStr, "%d", &usedFromUnix)
+	if parseErr != nil {
+		t.Logf(
+			"Warning: Malformed timestamp for node '%s' label '%s': %v. Treating as expired.",
+			nodeName,
+			NodeUsedFromLabel,
+			err,
+		)
+
+		return true
+	}
+
+	usedFrom := time.Unix(usedFromUnix, 0)
+	if time.Since(usedFrom) > expiryDuration {
+		t.Logf(
+			"Node '%s' previously used by '%s' has expired (used %s ago). Releasing.",
+			nodeName,
+			usedBy,
+			time.Since(usedFrom).Round(time.Second),
+		)
+
+		return true
+	}
+
+	return false
+}
+
+// LabelNodeAsUsed applies the usage labels to a node.
+func labelNodeAsUsed(ctx context.Context, client klient.Client, nodeName, testName string) error {
+	node, err := GetNodeByName(ctx, client, nodeName)
+	if err != nil {
+		return fmt.Errorf("failed to get node %s: %w", nodeName, err)
+	}
+
+	if node.Labels == nil {
+		node.Labels = make(map[string]string)
+	}
+
+	node.Labels[NodeUsedByLabel] = sanitizeTestName(testName)
```

Comment:
Since we are not using this for selection, can we use annotations instead? This would mean that we don't have to sanitize names